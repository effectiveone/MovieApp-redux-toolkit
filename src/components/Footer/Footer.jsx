import React from 'react'
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Link from '@mui/material/Link';
import uuid from "react-uuid";


export default function Footer() {
  return (
    <footer style={{ backgroundImage: `url(/bg.webp)` }}>
      <Box
        px={{ xs: 3, sm: 10 }}
        py={{ xs: 5, sm: 10 }}
        bgcolor="text.secondary"
        color="white"
      >
        <Container maxWidth="lg">
          <Grid container spacing={5}>
            <Grid item xs={12} sm={4}>
              <Box borderBottom={1}>Category</Box>
              {["Home", "Contact", "Terms of services", "About us"].map(z => (
                <Box key={uuid()}>
                  <Link href="/" color="inherit">
                    {z}
                  </Link>
                </Box>
              ))}
            </Grid>
            <Grid item xs={12} sm={4}>
              <Box borderBottom={1}>Account</Box>
              {[
                "Live",
                "FAQ",
                "Premium",
                "Privacy policy"
              ].map(z => (
                <Box key={uuid()}>
                  <Link href="/" color="inherit">
                    {z}
                  </Link>
                </Box>
              ))}
            </Grid>
            <Grid item xs={12} sm={4}>
              <Box borderBottom={1}>Messages</Box>
              {[
                "You must watch",
                "Recent release",
                "Top IMDB"].map(z => (
                  <Box key={uuid()}>
                    <Link href="/" color="inherit">
                      {z}
                    </Link>
                  </Box>
                ))
              }
            </Grid>
          </Grid>
          <Box textAlign="center" pt={{ xs: 5, sm: 10 }} pb={{ xs: 5, sm: 0 }}>
            MovieApp    &reg; {new Date().getFullYear()}
          </Box>
        </Container>
      </Box>
    </footer>
  );
}