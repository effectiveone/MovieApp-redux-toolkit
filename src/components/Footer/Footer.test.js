import React from "react";
import { render } from "@testing-library/react";
import Footer from "./Footer";
import { create } from "react-test-renderer";

describe("Footer", () => {
  it("renders the correct categories", () => {
    const { getByText } = render(<Footer />);
    expect(getByText("Category")).toBeInTheDocument();
    expect(getByText("Home")).toBeInTheDocument();
    expect(getByText("Contact")).toBeInTheDocument();
    expect(getByText("Terms of services")).toBeInTheDocument();
    expect(getByText("About us")).toBeInTheDocument();
  });

  it("renders the correct account links", () => {
    const { getByText } = render(<Footer />);
    expect(getByText("Account")).toBeInTheDocument();
    expect(getByText("Live")).toBeInTheDocument();
    expect(getByText("FAQ")).toBeInTheDocument();
    expect(getByText("Premium")).toBeInTheDocument();
    expect(getByText("Privacy policy")).toBeInTheDocument();
  });

  it("renders the correct message links", () => {
    const { getByText } = render(<Footer />);
    expect(getByText("Messages")).toBeInTheDocument();
    expect(getByText("You must watch")).toBeInTheDocument();
    expect(getByText("Recent release")).toBeInTheDocument();
    expect(getByText("Top IMDB")).toBeInTheDocument();
  });

  it("renders the correct trademark text", () => {
    const { asFragment, getByText } = render(<Footer />);
    expect(
      getByText(`MovieApp Â® ${new Date().getFullYear()}`)
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
