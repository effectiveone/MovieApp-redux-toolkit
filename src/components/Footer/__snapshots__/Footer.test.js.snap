// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the correct trademark text 1`] = `
<DocumentFragment>
  <footer
    style="background-image: url(/bg.webp);"
  >
    <div
      class="MuiBox-root css-aygekn"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-vcv939"
            >
              Category
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Contact
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Terms of services
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                About us
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-vcv939"
            >
              Account
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Live
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                FAQ
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Premium
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Privacy policy
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-vcv939"
            >
              Messages
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                You must watch
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Recent release
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Top IMDB
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1gjlakx"
        >
          MovieApp    Â® 2023
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
